src/Akka.Streams.Kafka/Stages/ConsumerStage.cs

```diff
@@ -117,6 +127,21 @@ public override void PostStop()
         private void HandleConsumeError(object sender, Message message)
         {
             Log.Error(message.Error.Reason);
+            var exception = new SerializationException(message.Error.Reason);
+            switch (_decider(exception))
+            {
+                case Directive.Stop:
+                    // Throw
+                    _completion.TrySetException(exception);
+                    FailStage(exception);
+                    break;
+                case Directive.Resume:
+                    // TODO: resume the stream
+                    break;
+                case Directive.Restart:
+                    // TODO: restart the stream
```

Comment:
@Silv3rcircl3 help me here

Reply:
Restart is mostly used in places where you have some initial state like StatefulSelect and on restart current state is replced with the initial one. I'm not sure if it makes sense here, if not you can just check for stop and otherwise ignore the excpetion.

Reply:
Also you maybe need to call Poll from here if Kafka isn't sending you new messages otherwise.